source/game_sa/CPad.cpp

```diff
@@ -41,287 +133,752 @@ void CPad::UpdateMouse() {
+    if (DisablePlayerControls) {
+        return 0;
+    }
+    switch (Mode) {
+    case 0:
+    case 1:
+    case 2:
+    case 3: {
+        SteeringLeftRightBuffer[0] = NewState.LeftStickX;
+        return SteeringLeftRightBuffer[DrunkDrivingBufferUsed];
+    }
+    }
+    return 0;
 }
 
 // Converted from thiscall short CPad::GetSteeringUpDown(void) 0x53FBD0
 short CPad::GetSteeringUpDown() {
-    return plugin::CallMethodAndReturn<short, 0x53FBD0, CPad*>(this);
+    if (DisablePlayerControls) {
+        return 0;
+    }
+    switch (Mode) {
+    case 0:
+    case 1:
+    case 2:
+    case 3:
+        return NewState.LeftStickY;
+    }
+    return 0;
 }
 
+// 0x540DC0
 short CPad::GetPedWalkLeftRight(CPed* pPed) {
-    return plugin::CallMethodAndReturn<short, 0x540DC0, CPad*, CPed*>(this, pPed);
+    if (DisablePlayerControls) {
+        return 0;
+    }
+    if (byte_B73403 || CCamera::m_bUseMouse3rdPerson) {
+        if (pPed && pPed->m_pAttachedTo) {
+            return 0;
+        } else if (byte_8CD782) {
+            return NewState.LeftStickX;
+        } else {
+            return NewState.RightStickX;
+        }
+    }
+    return 0;
 }
 
+// 0x540E20
 short CPad::GetPedWalkUpDown(CPed* pPed) {
-    return plugin::CallMethodAndReturn<short, 0x540E20, CPad*, CPed*>(this, pPed);
+    if (DisablePlayerControls) {
+        return 0;
+    }
+    if (byte_B73403 || CCamera::m_bUseMouse3rdPerson) {
+        if (pPed && pPed->m_pAttachedTo) {
+            return 0;
+        }
+        else if (byte_8CD782) {
+            return NewState.LeftStickY;
+        }
+        else {
+            return NewState.RightStickY;
+        }
+    }
+    return 0;
 }
 
 // Converted from thiscall short CPad::GetPedWalkLeftRight(void) 0x53FC90
 short CPad::GetPedWalkLeftRight() {
-    return plugin::CallMethodAndReturn<short, 0x53FC90, CPad*>(this);
+    if (DisablePlayerControls) {
+        return 0;
+    }
+    switch (Mode) {
+    case 0:
+    case 1:
+    case 2:
+    case 3:
+        return NewState.LeftStickX;
+    }
+    return 0;
 }
 
 // Converted from thiscall short CPad::GetPedWalkUpDown(void) 0x53FD30
 short CPad::GetPedWalkUpDown() {
-    return plugin::CallMethodAndReturn<short, 0x53FD30, CPad*>(this);
+    if (DisablePlayerControls) {
+        return 0;
+    }
+    switch (Mode) {
+    case 0:
+    case 1:
+    case 2:
+    case 3:
+        return NewState.LeftStickY;
+    }
+    return 0;
 }
 
 // Converted from thiscall bool CPad::GetLookLeft(void) 0x53FDD0
 bool CPad::GetLookLeft() {
-    return plugin::CallMethodAndReturn<bool, 0x53FDD0, CPad*>(this);
+    if (DisablePlayerControls) {
+        return 0;
+    }
+    if (NewState.LeftShoulder2 && !OldState.LeftShoulder2) {
+        return false;
+    } else if (!NewState.RightShoulder2 && OldState.RightShoulder2) {
+        return false;
+    } else {
+        return NewState.LeftShoulder2 && !OldState.RightShoulder2;
+    }
```

Comment:
Same as ``CPad::GetLookRight``.

```suggestion
    if (DisablePlayerControls)
        return false;

    if (NewState.LeftShoulder2 && !OldState.LeftShoulder2)
        return false;

    if (!NewState.RightShoulder2 && OldState.RightShoulder2)
        return false;

    return NewState.LeftShoulder2 && !NewState.RightShoulder2;
```

Reply:
Last line differs from my code.

Reply:
![image](https://user-images.githubusercontent.com/1680033/128104082-bb9a8762-8756-4f9b-8e98-c8cc65dccbe1.png)

Reply:
Nice, thanks!